### PR TITLE
test(views_integration) update, fix, re-enable

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3388

Based on: https://github.com/moderntribe/tribe-testing-facilities/pull/24

This PR fixes, extends and updates our test code to cover latest past events views.

The fixes consist in the usage of up-to-date base test cases, from `tribe-testing-facilities`,
to ensure the starting context of each test is as clean as we can get it. During this work
I've found a number of instances where snapshots genereated while running the tests alone (i.e.
running on test method only to regenerate its snapshots) would, then, fail when running the
same test in the context of a whole suite.

I've found pretty good direct coverage of the latest past events in the code and added some for
List, Day and Month View.

### CODE REVIEW

File: src/Tribe/Service_Providers/Context.php [hunk 1]
```diff
@@ -378,6 +379,22 @@ static function () {
 						Tribe__Context::QUERY_VAR        => [ 'tribe_view', 'eventDisplay' ],
 					],
 				],
+				'latest_event_date'    => [
+					'read'  => [
+						Tribe__Context::TRIBE_OPTION => [ 'latest_date' ]
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php:384]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: b52bad456e88a5f856e0c8ab0d450b7f</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 2]
```diff
@@ -378,6 +379,22 @@ static function () {
...
+					],
+					'write' => [
+						Tribe__Context::TRIBE_OPTION => [ 'latest_date' ]
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php:387]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: b95029b65684d8870d1878dd58d44933</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 3]
```diff
@@ -378,6 +379,22 @@ static function () {
...
+					],
+				],
+				'earliest_event_date'  => [
+					'read'  => [
+						Tribe__Context::TRIBE_OPTION => [ 'earliest_date' ]
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php:392]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: d66a793d21e08c32c9f749bfcd0dcd8d</sub>

File: src/Tribe/Service_Providers/Context.php [hunk 4]
```diff
@@ -378,6 +379,22 @@ static function () {
...
+					],
+					'write' => [
+						Tribe__Context::TRIBE_OPTION => [ 'earliest_date' ]
```
Comment (tr1b0t) [inline src/Tribe/Service_Providers/Context.php:395]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 8d129b78152924b9df18cde876492fa7</sub>

File: composer.json [hunk 5]
```diff
@@ -25,7 +25,7 @@
     "lucatume/wp-snaphot-assertions": "1.1.0",
     "mikey179/vfsstream": "^1.6",
     "moderntribe/tribalscents": "dev-master",
-    "moderntribe/tribe-testing-facilities": "dev-master",
+    "moderntribe/tribe-testing-facilities": "dev-fix/views-v2-better-reset",
```
Comment (Camwyn) [inline composer.json:28]: Do you want to merge the other PR first and change this back?

File: tests/views_integration/Tribe/Events/Views/V2/ExtendingViewTest.php [hunk 6]
```diff
@@ -111,14 +114,23 @@ public function should_fetch_the_template_from_the_theme_if_available() {
 	/**
 	 * @test
 	 */
-	public function should_render_the_not_found_view_if_trying_to_render_not_registered_view() {
-		$this->markTestSkipped( 'Skipping the 404 since we need to revist the behavior here.' );
-
-		$view = View::make( 'not-set' );
+	public function should_render_the_default_view_if_view_not_found() {
```
Comment (Camwyn) [inline tests/views_integration/Tribe/Events/Views/V2/ExtendingViewTest.php:117]: Does this reflect a change in behavior?

File: composer.json (hunk 5 above)
Comment (lucatume) [reply composer.json:28]: yes, I've updated in lates commit.

File: tests/views_integration/Tribe/Events/Views/V2/ExtendingViewTest.php (hunk 6 above)
Comment (lucatume) [reply tests/views_integration/Tribe/Events/Views/V2/ExtendingViewTest.php:117]: Yes, we used to print a "not found" view template, we now print the default view template.

**Pull request merged**